test/Microsoft.Extensions.DataIngestion.Tests/Chunkers/SemanticChunkerTests.cs

```diff
@@ -74,8 +71,121 @@ public async Task TwoTopicsParagraphs()
+| Hera | Marriage & Family | Peacock | Juno |
+| Poseidon | Sea & Earthquakes | Trident | Neptune |
+| Athena | Wisdom & War | Owl | Minerva |
+| Apollo | Sun & Music | Lyre | Apollo |
+| Artemis | Hunt & Moon | Silver Bow | Diana |
+| Aphrodite | Love & Beauty | Dove | Venus |
+| Ares | War & Courage | Spear | Mars |
+| Hephaestus | Fire & Forge | Hammer | Vulcan |
+| Demeter | Harvest & Nature | Wheat | Ceres |
+| Dionysus | Wine & Festivity | Grapes | Bacchus |
+| Hermes | Messages & Trade | Caduceus | Mercury |";
+
+            Document doc = new("dotnet-languages");
+            doc.Sections.Add(new DocumentSection
+            {
+                Elements =
+                {
+                    new DocumentHeader("# .NET Supported Languages") { Level = 1 },
+                    new DocumentParagraph("The .NET platform supports multiple programming languages:"),
+                    new DocumentTable(dotNetTableMarkdown, CreateLanguageTableCells()),
+                    new DocumentParagraph("C# remains the most popular language for .NET development."),
+                    new DocumentHeader("# Ancient Greek Olympian Gods") { Level = 1 },
+                    new DocumentParagraph("The twelve Olympian gods were the principal deities of the Greek pantheon:"),
+                    new DocumentTable(godsTableMarkdown, CreateGreekGodsTableCells()),
+                    new DocumentParagraph("These gods resided on Mount Olympus and ruled over different aspects of mortal and divine life.")
+                }
+            });
+
+            IDocumentChunker chunker = CreateDocumentChunker(maxTokensPerChunk: 200);
+            List<DocumentChunk> chunks = await chunker.ProcessAsync(doc);
+            
+            Assert.Equal(3, chunks.Count);
+            Assert.Equal($@"# .NET Supported Languages
+The .NET platform supports multiple programming languages:
+{dotNetTableMarkdown}
+C# remains the most popular language for .NET development."
+            , chunks[0].Content, ignoreLineEndingDifferences: true);
+            Assert.Equal($@"# Ancient Greek Olympian Gods
+The twelve Olympian gods were the principal deities of the Greek pantheon:
+| God | Domain | Symbol | Roman Name |
+| --- | --- | --- | --- |
+| Zeus | Sky & Thunder | Lightning Bolt | Jupiter |
+| Hera | Marriage & Family | Peacock | Juno |
+| Poseidon | Sea & Earthquakes | Trident | Neptune |
+| Athena | Wisdom & War | Owl | Minerva |
+| Apollo | Sun & Music | Lyre | Apollo |
+| Artemis | Hunt & Moon | Silver Bow | Diana |
+| Aphrodite | Love & Beauty | Dove | Venus |
+| Ares | War & Courage | Spear | Mars |
+| Hephaestus | Fire & Forge | Hammer | Vulcan |
+| Demeter | Harvest & Nature | Wheat | Ceres |
+| Dionysus | Wine & Festivity | Grapes | Bacchus |"
+            , chunks[1].Content, ignoreLineEndingDifferences: true);
+            Assert.Equal($@"| God | Domain | Symbol | Roman Name |
+| --- | --- | --- | --- |
+| Hermes | Messages & Trade | Caduceus | Mercury |
```

Comment:
this test  shows what the `SemanticChunker` has benefited from this PR: the ability to split large tables